src/liballoc/sync/tests.rs

```diff
@@ -337,20 +336,25 @@ fn test_ptr_eq() {
 
 #[test]
 #[cfg_attr(target_os = "emscripten", ignore)]
-#[cfg_attr(miri, ignore)] // Miri does not support threads
 fn test_weak_count_locked() {
     let mut a = Arc::new(atomic::AtomicBool::new(false));
     let a2 = a.clone();
     let t = thread::spawn(move || {
-        for _i in 0..1000000 {
+        // Miri is too slow
+        let count = if cfg!(miri) { 100 } else { 1000000 };
+        for _i in 0..count {
             Arc::get_mut(&mut a);
+            #[cfg(miri)] // The Miri scheduler won't reschedule if we do not yield.
+            thread::yield_now();
         }
         a.store(true, SeqCst);
     });
 
     while !a2.load(SeqCst) {
         let n = Arc::weak_count(&a2);
         assert!(n < 2, "bad weak count: {}", n);
+        #[cfg(miri)] // The Miri scheduler won't reschedule if we do not yield.
+        thread::yield_now();
     }
```

Comment:
@vakaras this demonstrates an issue with the current Miri scheduling policy: I had to adjust the code to avoid deadlocks. IMO we should introduce something where threads also get descheduled occasionally even when they do not yield.

Reply:
> I had to adjust the code to avoid deadlocks

If I understand correctly, Miri did not terminate in this case, right?

> IMO we should introduce something where threads also get descheduled occasionally even when they do not yield.

I have to agree with you.

Reply:
> If I understand correctly, Miri did not terminate in this case, right?

Yes. It just ran and ran and probably was just sitting in this loop forever.